fix(DB/smart_scripts): Quest "Enlistment Day"

FALL1N1 opened the pull request:
<!-- First of all, THANK YOU for your contribution.
 Please fill this template unless your PR is very simple/straightforward.
 Do not forget to have a look at our Pull Request tutorial: http://www.azerothcore.org/wiki/Contribute#how-to-create-a-pull-request
-->

<!-- WRITE A RELEVANT TITLE -->

## CHANGES PROPOSED:
-  
-  


## ISSUES ADDRESSED:
- Closes #3699
<!-- If the issue doesn't exist, describe it and how to reproduce it, please. If the issue already exists, just paste the link to the issue you close, like this: Closes https://github.com/azerothcore/azerothcore-wotlk/issues/967 -->


## TESTS PERFORMED:
- http://prntscr.com/vlzfnw
-
<!-- Does it build without errors? Did you test in-game? What did you test? Did you do all these tests on Linux, Mac or Windows? Other tests performed -->


## HOW TO TEST THE CHANGES:
<!-- We need to confirm the changes first, so try to make the work easy for testers (who are not necessarily coders), please:
 - Which commands to use? Which NPC to teleport to?
 - Do we need to enable debug flags on Cmake?
 - Do we need to look at the console? etc...
 - Other steps
-->


## KNOWN ISSUES AND TODO LIST:
<!-- This is a TODO list with checkboxes to tick -->
- [ ]
- [ ] 


## Target branch(es):
- [x] Master


<!-- NOTE: You do not need to squash your commits, on merge we will squash them for you (when there are too many commits we merge them into one big commit for a cleaner and easy to read history). -->

<!-- NOTE2: If you intend to contribute more than once, you should really join us on our discord channel!
 The link is on our site http://azerothcore.org/ We set cosmetic ranks for our contributors and may give access to special resources/knowledge to them! -->


<!-- Do not remove the instructions below about testing, they will help users to test your PR -->
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

Voxstrasza commented:
thank you fallin! you're amazing, I tested this out asap and here's my feedback:

This happens upon server start/loading the map (empty space): https://i.imgur.com/mi9JzhI.jpg - but when new ones spawn and get into the line, they move up to begin the scripted event, and the NPCs in the 'static' line (the one that isn't supposed to move) aren't supposed to have weapons in their hand

Jumping on table: https://i.imgur.com/MTMRZaF.jpg, can this be moved back a bit so the NPCs don't jump on the table?

No dialogue after turning in Enlistment Day: https://i.imgur.com/OQWMBS0.jpg - I did get dialogue after accepting the quest though! (first screenshot)

Voxstrasza commented:
Worldserver errors:

```
2020-11-19 14:52:52 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:52:54 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:52:58 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:00 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:04 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
2020-11-19 14:53:06 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
2020-11-19 14:53:30 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:53:34 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 1 for talker 25317, ignored.
2020-11-19 14:53:36 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:40 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 10 using non-existent Text id 2 for talker 25317, ignored.
2020-11-19 14:53:42 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11778800 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
2020-11-19 14:53:46 ERROR: SmartScript::ProcessAction: SMART_ACTION_TALK: EntryOrGuid 11779400 SourceType 9 EventType 1 TargetType 1 using non-existent Text id 3 for talker 25317, ignored.
```

FALL1N1 commented:
Looks like we are missing texts, I'll re-check it tomorrow, thank you!

Maurowin commented:
Do you will adds all dial for every location ?

FALL1N1 commented:
> 
> 
> Do you will adds all dial for every location ?

what do you mean?

FALL1N1 commented:
Added misc texts
Fixed the waypoint being on the table instead of in front of it (they still walk over a bit, but that's from sniffs, idk if it's ok to replace with custom WP while it'd indeed be better)
The NPCS are initially missing, that's blizzlike as of the SAI timers. Once there's a player around the event will automatically start and they will end up in a queue.

Maurowin commented:
> 
> 
> > Do you will adds all dial for every location ?
> 
> what do you mean?

add all language, ES, FR, DE etc... for all dialogue

FALL1N1 commented:
No, just English, I do not have sniffs as I am backporting this from Trinity.

Voxstrasza commented:
> 
> 
> Added misc texts
> Fixed the waypoint being on the table instead of in front of it (they still walk over a bit, but that's from sniffs, idk if it's ok to replace with custom WP while it'd indeed be better)
> The NPCS are initially missing, that's blizzlike as of the SAI timers. Once there's a player around the event will automatically start and they will end up in a queue.

https://i.imgur.com/H5AxUaC.jpg / https://i.imgur.com/konH2dX.jpg - up to you if you want to change it, I think it looks good.

https://i.imgur.com/GGGkqEU.jpg - works, civilian recruit practicing sword in the basement talks too much though

great work!! 💯

FALL1N1 commented:
@Voxstrasza it's fine by me, although it has to be approved by a team member.

FrancescoBorzi commented on data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:
```diff
@@ -0,0 +1,144 @@
+DELETE FROM `creature_text` WHERE `CreatureID` in(25317,25220,25222);
```
please use uppercase `IN` and add a space after it

FrancescoBorzi commented on data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:
```diff
@@ -0,0 +1,144 @@
+(25220, 9, 2289.469, 5244.898, 11.45096, 'Civilian Recruit'),
+(25220, 10, 2291.772, 5243.933, 11.45096, 'Civilian Recruit'),
+(25220, 11, 2294.129, 5242.708, 11.45096, 'Civilian Recruit'),
+-- (25220, 12, 2298.062, 5241.932, 12.3176, 'Civilian Recruit'),
```
?

FALL1N1 replied:
Left for reference, the commented out point12 is blizzlike (they end up standing on the table), the custom point 12 is a bit back, so they don't do that. I have left a comment.

FrancescoBorzi commented on data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:
```diff
@@ -0,0 +1,144 @@
+UPDATE `creature_template` SET `AIName`='SmartAI' WHERE  `entry`IN(24959,25220,25307);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid`IN(-107575,-107574);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid`IN(25220,25307);
+DELETE FROM `smart_scripts` WHERE `source_type`=9 AND `entryorguid`IN(2522000,2522001);
```
space after `IN`

FrancescoBorzi commented on data/sql/updates/pending_db_world/0000_00_00_00_lolz.sql:
```diff
@@ -0,0 +1,144 @@
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@GUID AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`tar ...
+(@GUID,0,0,0,1,0,100,0,8000,8000,12000,12000,5,36,0,0,0,0,0,1,0,0,0,0,0,0,0,"Civilian Recruit - Out of Combat - Play Emote 36"),
+(@GUID,0,1,0,1,0,100,0,35000,40000,35000,40000,80,11779400,0,0,0,0,0,1,0,0,0,0,0,0,0,"Civilian Recruit - Out of Combat - Run Script");
```
please use single quotes for strings

Voxstrasza commented:
can the swords be removed on the Civilian Recruits in the line that doesn't move, just to make it a bit more blizzlike?

Voxstrasza commented:
@FALL1N1 is this done?

FALL1N1 commented:
@Voxstrasza yes, it is.

FALL1N1 commented:
merged with master, updated and good to go

Voxstrasza commented:
all good on my end!

Helias commented:
Thanks for the test!

So, @Voxstrasza from your point of view we can put this as "To Be Merged"?

Voxstrasza commented:
yes it works beautifully

Si1ker commented on data/sql/updates/pending_db_world/rev_1607432767438370200.sql:
```diff
@@ -0,0 +1,149 @@
+DELETE FROM  `creature` WHERE `guid` IN (108008,108007,108006,108005,108004,108003,108002,108001,108000);
+
+UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry` IN (24959,25220,25307);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid` IN (-107575,-107574);
+DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid` IN (25220,25307);
```
@FALL1N1 ?
```suggestion
DELETE FROM `smart_scripts` WHERE `source_type`=0 AND `entryorguid` IN (-107575 ,-107574, 25220, 25307);
```

Si1ker commented on data/sql/updates/pending_db_world/rev_1607432767438370200.sql:
```diff
@@ -0,0 +1,149 @@
+(25317, 3, 2, 'I\'m just nervous.  Sorry.', 12, 7, 100, 0, 0, 0, 26493, 'Civilian Recruit');
+
+DELETE FROM `creature_equip_template` WHERE `CreatureID` IN (25317, 25220);
+UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;
```
new line at the end
```suggestion
UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;

```

Si1ker commented on data/sql/updates/pending_db_world/rev_1607432767438370200.sql:
```diff
@@ -0,0 +1,148 @@
+(25317, 3, 2, 'I\'m just nervous.  Sorry.', 12, 7, 100, 0, 0, 0, 26493, 'Civilian Recruit');
+
+DELETE FROM `creature_equip_template` WHERE `CreatureID` IN (25317, 25220);
+UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;
```
```suggestion
UPDATE `creature` SET `equipment_id`=0 WHERE `id` = 25317;


```
